app/models/manageiq/providers/hawkular/middleware_manager/middleware_deployment.rb

```diff
@@ -1,4 +1,14 @@
 module ManageIQ::Providers
   class Hawkular::MiddlewareManager::MiddlewareDeployment < MiddlewareDeployment
+    def self.resource_path_for_metrics(item)
+      path = ::Hawkular::Inventory::CanonicalPath.parse(item.ems_ref)
```

Comment:
Yeah, sure, I've been writing some stuff down for the past few weeks, and plan to release as soon as I have something concrete. About this one, I think that's not exactly a problem, and can be changed ad-hoc.